docs/generator_template.md

```diff
@@ -0,0 +1,569 @@
+- You'll use [Python Paho-MQTT](https://pypi.org/project/paho-mqtt/) as the **MQTT client** in this project.
+- Lastly, you will create a React template that will use the MQTT broker to allow you to monitor your bedroom's temperature and notify you when the temperature drops or rises above 22 °C.
+- Create a reusable component for the output code's rise/dropped temperature functions.
+
+## Background context
+
+There is a list of [community maintained templates](https://www.asyncapi.com/docs/tools/generator/template#generator-templates-list), but what if you require customized output from the generator? In that case, you'll create a user-defined template that generates custom output from the generator.
+Before you create the template, you'll need to have an [AsyncAPI document](https://www.asyncapi.com/docs/tools/generator/asyncapi-document) that defines the properties you want to use in your template to test against. In this tutorial, you'll use the following template saved in the `test/fixtures/asyncapi.yml` file in your template project directory.
+
+``` yml
+
+asyncapi: 2.6.0
+
+info:
+  title: Temperature Service
+  version: 1.0.0
+  description: This service is in charge of processing all the events related to temperature.
+
+servers:
+  dev:
+    url: test.mosquitto.org
+    protocol: mqtt
+    bindings:
+      mqtt:
+        clientId: temperature-service
+
+channels:
+  temperature/changed:
+    description: Updates the bedroom temperature in the database when the temperatures drops or goes up.
+    publish:
+      operationId: temperatureChange
+      message:
+        description: Message that is being sent when the temperature in the bedroom changes.
+        payload:
+          type: object
+          additionalProperties: false
+          properties:
+            temperatureId:
+              type: string
+components:
+  schemas:
+    temperatureId:
+      type: object
+      additionalProperties: false
+      properties:
+        temperatureId:
+          type: string
+```
+
+Note:
+
+- To generate code, use the [AsyncAPI CLI](https://www.asyncapi.com/tools/cli). If you don't have the CLI installed, follow [this guide](https://www.asyncapi.com/docs/tools/generator/installation-guide#asyncapi-cli).
+- If you are new to AsyncAPI, check out the following docs: [template development](https://www.asyncapi.com/docs/tools/generator/template-development), which explains the minimum requirements for a template and possible features.
+- You can [learn more about asynchronous messaging using MQTT](https://medium.com/python-point/mqtt-basics-with-python-examples-7c758e605d4).
+
+## Steps followed to create a template
+
+1. Create a new directory for your template named **python-mqtt-client-template**.
+2. Install the AsyncAPI CLI using the command `npm install -g @asyncapi/cli`.
+3. Create a new folder **test/fixtures** with a file named `asyncapi.yml` in your fixtures directory. This file is used to define the **structure** of your template. 
```

Comment:
should we maybe refer to backgroud context section? like tell people that `asyncapi.yml` should contain content mentioned there?